src/Output.cxx

```diff
@@ -748,14 +748,9 @@ ASTVisitor::DumpId ASTVisitor::AddTypeDumpNode(DumpType dt, bool complete,
         DumpType(t->getAs<clang::DecayedType>()->getDecayedType(), c),
         complete, dq);
     case clang::Type::Decltype:
-      if (this->Opts.CastXml && t->isNullPtrType()) {
-        clang::DecltypeType const* dtt = t->getAs<clang::DecltypeType>();
-        if (dtt->getUnderlyingExpr()->getStmtClass() ==
-            clang::Stmt::CXXNullPtrLiteralExprClass) {
-          // Treat literal 'decltype(nullptr)' as a FundamentalType.
-          return this->AddTypeDumpNode(DumpType(dtt->getUnderlyingType(), c),
-                                       complete, dq);
-        }
+      if (this->Opts.CastXml) {
+        return this->AddTypeDumpNode(DumpType(t.getCanonicalType(), c),
+                                     complete, dq);
```

Comment:
Please use `getUnderlyingType()` if it works.  It's more explicitly associated with `clang::DecltypeType`.

Reply:
Thanks! This seems to work well for our use case. Sorry for not using that earlier...